components/watcher/src/watcher.rs

```diff
@@ -145,15 +142,36 @@ impl WatcherData {
                     });
                 }
 
-                let result = process_signed_query(qry.clone(), &self.event_storage);
-                let response = match result {
-                    Err(SignedQueryError::QueryError(QueryError::UnknownId { .. })) => {
-                        // forward query to witness
-                        self.forward_query(&qry).await?;
-                        return Ok(None);
-                    }
-                    result => result?,
-                };
+                // Update latest state for prefix
+                let wit_id = self.query_state(qry.query.get_prefix()).await?;
+
+                // Get latest state for prefix
+                let last_sn = self
+                    .event_storage
+                    .get_last_ksn_reply(&qry.query.get_prefix(), &IdentifierPrefix::Basic(wit_id))
+                    .and_then(|ksn| {
+                        if let ReplyRoute::Ksn(_, ksn) = ksn.reply.get_route() {
+                            Some(ksn)
+                        } else {
+                            None
+                        }
+                    })
+                    .map(|ksn| ksn.state.sn);
+
+                // Compute current state from our database
+                let current_sn = self
+                    .event_storage
+                    .get_ksn_for_prefix(&qry.query.get_prefix(), SerializationFormats::JSON)?
+                    .state
+                    .sn;
+
+                if current_sn < last_sn.unwrap_or(0) {
```

Comment:
This is always true. Reply is saved as accepted only if there are enough events in event storage. If one gets ksn message with sn newer than he has in local kel, it is escrowed and it means that he is waiting for missing events. So maybe we should use ksn from escrow to compute `last_sn`?